src/core/qgstransaction.cpp

```diff
@@ -81,6 +81,46 @@ QgsTransaction::~QgsTransaction()
   setLayerTransactionIds( nullptr );
 }
 
+// For the needs of the OGR provider with GeoPackage datasources, remove
+// any reference to layers in the connection string
+QString QgsTransaction::removeLayerIdOrName( const QString &str )
```

Comment:
That reminds me, I always wondered how to extract the layer name part from ogr connection strings, I don't think we ship something like this in QDataSourceUri?
(totally off-topic ;) )

Reply:
Yes, that's a bit annoying that OGR layer connection string is a completely custom format not taking advantage of the parameter separation offered by QgsDataSourceUri

Reply:
Do you think an integration is not possible?
If not, we could at least introduce a QgsOgrUri at some point that encapsulates the parsing.

Reply:
I guess it could go to the common scheme, but that's more a matter of backward compatibility for people using projects from older versions.
Yes a QgsOgrUri could be a compromise

Reply:
As far as I can see it's mostly `uri()` which is returned at the moment and `table()` and co never return anything. If it's possible to modify things so, that only `table()` and co methods return something and the rest stays stable, that would not bear any risk concerning backwards compatibility.